File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cs
public class Solution {
    public string ReverseVowels(string s) {
        int l = 0;
        int r = s.Length - 1;

        HashSet<char> vowels = new HashSet<char>(){
            'a','e','i','o','u',
            'A','E','I','O','U'
        };

        char[] arr = s.ToCharArray();
        while(l < r){
            while(l < r && !vowels.Contains(arr[l])){
                l++;
            }
            while(l < r && !vowels.Contains(arr[r])){
                r--;
            }
            if(l < r){
                char temp = arr[l];
                arr[l] = arr[r];
                arr[r] = temp;
                
                r--;
                l++;
            }
        }
        return new string(arr);
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function (s) {
    let l = 0;
    let r = s.length - 1;
    s = s.split("")
    let vowels = ['a', 'e', 'i', 'o', 'u']
    while (l < r) {
        if (vowels.includes(s[l].toLowerCase()) && vowels.includes(s[r].toLowerCase())) {
            let temp = s[l]
            s[l] = s[r]
            s[r] = temp
            l++;
            r--;

        }
        else if (vowels.includes(s[l].toLowerCase()) && !vowels.includes(s[r].toLowerCase())) {
            r--
        }
        else {
            l++
        }
    }
    return s.join("")

};